app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt

```diff
@@ -1,5 +1,130 @@
+    }
+
+    override fun onItemClicked(transaction: HistoryTransaction) {
+        launch {
+            when (transaction) {
+                is HistoryTransaction.Swap -> {
+                    swapAnalytics.logSwapShowingDetails(
+                        swapStatus = SwapAnalytics.SwapStatus.SUCCESS,
+                        lastScreenName = analyticsInteractor.getPreviousScreenName(),
+                        tokenAName = transaction.sourceSymbol,
+                        tokenBName = transaction.destinationSymbol,
+                        swapSum = transaction.amountA,
+                        swapUSD = transaction.amountSentInUsd ?: BigDecimal.ZERO,
+                        feesSource = SwapAnalytics.FeeSource.UNKNOWN
+                    )
+                }
+                is HistoryTransaction.Transfer -> {
+                    val renBtcSession = renBtcInteractor.findActiveSession()
+                    val isRenBtcSessionActive = renBtcSession != null && renBtcSession.isValid
+
+                    if (transaction.isSend) {
+                        val sendNetwork =
+                            if (isRenBtcSessionActive) {
+                                SendAnalytics.SendNetwork.BITCOIN
+                            } else {
+                                SendAnalytics.SendNetwork.SOLANA
+                            }
+                        sendAnalytics.logSendShowingDetails(
+                            sendStatus = SendAnalytics.SendStatus.SUCCESS,
+                            lastScreenName = analyticsInteractor.getPreviousScreenName(),
+                            tokenName = transaction.tokenData.symbol,
+                            sendNetwork = sendNetwork,
+                            sendSum = transaction.total,
+                            sendUSD = transaction.totalInUsd ?: BigDecimal.ZERO
+                        )
+                    } else {
+                        val receiveNetwork =
+                            if (isRenBtcSessionActive) {
+                                ReceiveAnalytics.ReceiveNetwork.BITCOIN
+                            } else {
+                                ReceiveAnalytics.ReceiveNetwork.SOLANA
+                            }
+                        receiveAnalytics.logReceiveShowingDetails(
+                            receiveSum = transaction.total,
+                            receiveUSD = transaction.totalInUsd ?: BigDecimal.ZERO,
+                            tokenName = transaction.tokenData.symbol,
+                            receiveNetwork = receiveNetwork
+                        )
+                    }
+                }
+                else -> {
+                    // TODO: Add support for other transaction types
+                    // do nothing yet
+                    return@launch
+                }
+            }
+
+            view?.openTransactionDetailsScreen(transaction)
+        }
+    }
+
+    private fun onSuccess(items: List<HistoryTransaction>) {
```

Comment:
onTransactionsLoadedSucess

we can onSuccess for different actions, so it's better to be explicit about it